src/wallet/rpcwallet.cpp

```diff
@@ -3100,24 +3103,40 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
     if (!EnsureWalletIsAvailable(fHelp))
         return NullUniValue;
 
-    if (fHelp || params.size() > 0)
+    std::string defaultType = ADDR_TYPE_SPROUT;
+
+    if (fHelp || params.size() > 1)
         throw runtime_error(
-            "z_getnewaddress\n"
-            "\nReturns a new zaddr for receiving payments.\n"
+            "z_getnewaddress ( type )\n"
+            "\nReturns a new shielded address for receiving payments.\n"
+            "\nWith no arguments, returns a Sprout address.\n"
             "\nArguments:\n"
+            "1. \"type\"         (string, optional, default=\"" + defaultType + "\") The type of address. One of [\""
+            + ADDR_TYPE_SPROUT + "\", \"" + ADDR_TYPE_SAPLING + "\"].\n"
             "\nResult:\n"
-            "\"zcashaddress\"    (string) The new zaddr\n"
+            "\"zcashaddress\"    (string) The new shielded address.\n"
             "\nExamples:\n"
             + HelpExampleCli("z_getnewaddress", "")
+            + HelpExampleCli("z_getnewaddress", ADDR_TYPE_SAPLING)
             + HelpExampleRpc("z_getnewaddress", "")
```

Comment:
Why do we update ```HelpExampleCli```, but not ```HelpExampleRpc```?

Reply:
I just didn't add an example, because you can conceptually reuse the CLI examples inside the RPC ones (and pretty much everywhere else only one RPC example is given).